DataFormats/Detectors/TRD/include/DataFormatsTRD/LinkRecord.h

```diff
@@ -29,35 +29,56 @@ namespace trd
 class LinkRecord
 {
   using DataRange = o2::dataformats::RangeReference<int>;
-
  public:
   LinkRecord() = default;
-  LinkRecord(const uint32_t hcid, int firstentry, int nentries) : mLinkId(hcid), mDataRange(firstentry, nentries) {}
+  LinkRecord(const uint32_t linkid, int firstentry, int nentries) : mLinkId(linkid), mDataRange(firstentry, nentries) {}
+  LinkRecord(uint32_t sector, int stack, int layer, int side, int firstentry, int nentries) : mDataRange(firstentry, nentries) { setLinkId(sector, stack, layer, side); }
+
   ~LinkRecord() = default;
 
   void setLinkId(const uint32_t linkid) { mLinkId = linkid; }
+  void setLinkId(const uint32_t sector, const uint32_t stack, const uint32_t layer, const uint32_t side);
   void setDataRange(int firstentry, int nentries) { mDataRange.set(firstentry, nentries); }
   void setIndexFirstObject(int firstentry) { mDataRange.setFirstEntry(firstentry); }
   void setNumberOfObjects(int nentries) { mDataRange.setEntries(nentries); }
 
   uint32_t getLinkId() { return mLinkId; }
   uint32_t getLinkHCID() { return mLinkId & 0x7ff; } // the last 11 bits.
+  uint32_t getSector() { return (mLinkId & sectorbp) << sectorbs; }
+  uint32_t getStack() { return (mLinkId & stackbp) << stackbs; }
+  uint32_t getLayer() { return (mLinkId & layerbp) << layerbs; }
+  uint32_t getSide() { return (mLinkId & sidebp) << sidebs; }
   int getNumberOfObjects() const { return mDataRange.getEntries(); }
   int getFirstEntry() const { return mDataRange.getFirstEntry(); }
+  static uint32_t getHalfChamberLinkId(uint32_t detector, uint32_t rob);
+  static uint32_t getHalfChamberLinkId(uint32_t sector, uint32_t stack, uint32_t layer, uint32_t side);
+
+  void printStream(std::ostream& stream);
 
-  void printStream(std::ostream& stream) const;
+  static constexpr int sectorbp = 0xf80;
```

Comment:
Wouldn't this be simpler with a struct?

It's a matter of style, not a request to change anything.

Reply:
Ja a struct would be better and probably more elegant. 

Will change later. Unless you want it changed  now.

Reply:
changed to a struct.